source/develop/release-management/features/network/manageiq_ovn.html.md

```diff
@@ -0,0 +1,140 @@
+* Manual refresh of oVirt's network provider.
+* Automatic refresh of oVirt's network provider (I'm not sure if it works for OpenStack- Neutron)
+
+## Limitations
+* ManageIQ supports only one network provider for oVirt.
+* In oVirt there is no way to distinguish between Openstack network provider and OVN provider. Therefore, ManageIQ has no way to find out what provider is the OVN one.
+In case oVirt has more than one network provider, the first one will be used.
+* ManageIQ is using the first network provider returned by /ovirt-engine/api/openstacknetworkproviders. If there is more than one, it can cause strange behavior in ManageIQ (I'm not sure, but I guess the order of the returned network providers is not persistent. Therefore, each time ManageIQ will read the details of another provider).
+* Currently, ManageIQ doesn't support removal of the network provider.
+
+## Benefit to oVirt
+* The ui and the communication code with the network provider is already written in ManageIQ (for OpenStack-Neutron). Adding it from scratch to oVirt will take much more time than reusing the code of ManageIQ,
+
+## Benefit to ManageIQ
+* Adding more power to the oVirt's infra manager.
+* oVirt users that want to use OVN provider (more than the basic operations oVirt supports) will have to use ManageIQ.
+
+## Gaps with the OpenStack network provider
+
+### Gaps
+* Communication and authentication
+  * The OpenStack infra manager added to ManageIQ has the Keystone URL. OpenStack infra manager discovers the other OpenStack services via Keystone. The Keystone returns a service catalog with all the URLs and other information about the services. This information is used to communicate with the other services.
+  * The network manager is an ExtManagementSystem.
+  * OpenStack's network manager delegates all its communication and authentication methods to the parent manager (infra manager => keystone).
+  * The oVirt's infra manager is communicating with oVirt but the network manager should communicate directly with OVN provider. Therefore, the delegation the OpenStack's network manager has is problematic for oVirt.
+* Service Catalog
+  * In oVirt when adding a new network provider, the user supplies two values - `provider_url` and `authentication_url` (the URL of Keystone).
+First an authentication is done against Keystone to get the `token` and then oVirt communicates with the provider using the `provider_url` (supplying the `token` for authentication).
+  * In ManageIQ the 'services' URLs are not supplied by the user but determined from the Keystone. When Keystone is asked for the `token` it replies with the `token` and a `service_catalog` with all the services information (URLs, credentials, etc).
+* Tenants
+  * Neutron supports tenants.
+  * oVirt's OVN provider doesn't.
+  * ManageIQ expects the Neutron network provider to have tenants.
+* oVirt's OVN provider has 'v2' Keystone API. However, 'v2' API is already considered as legacy.
+
```

Comment:
One more gap could be adding an authentication plugin to the provider.
Currently the provider authorizes users through ovirt

Reply:
Added to the follow up features.